### PR TITLE
Separate liveness and readiness probes

### PR DESCRIPTION
**Pull request by ryan-hansen**
Allow for separate paths and settings for liveness and readiness probes.

### CODE REVIEW

File: charts/service/templates/deployment.yaml [hunk 1]
```diff
@@ -112,16 +112,16 @@ spec:
               protocol: TCP
           livenessProbe:
             httpGet:
-              path: {{ .Values.health.path | quote }}
+              path: {{ .Values.health.liveness.path | quote }}
```
Comment (paraplexed) [inline charts/service/templates/deployment.yaml:115]: This is a breaking change. Do we want to support the old way as a fallback if they don't have `health.liveness` set?

Comment (ryan-hansen) [reply charts/service/templates/deployment.yaml]: Good call.  Updated for backward compatibility.

File: charts/service/templates/_probes.tpl [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+{{- define "service.probes" }}
+livenessProbe:
+  httpGet:
+    {{- if .Values.health.liveness }}
```
Comment (paraplexed) [inline charts/service/templates/_probes.tpl:5]: Should we swap this to `if .Values.health.path` instead? Then we have people remove `health.path` as part of the deprecation strategy. Basically like an "opt-in" strategy to this, until we officially deprecate it by doing a breaking change version bump.

File: charts/service/values.yaml [hunk 3]
```diff
@@ -48,12 +48,17 @@ service:
...
+# WARNING: The following properties will be deprecated in upcoming versions of the helm chart:
+#   health.path
+#   health.initialDelaySeconds
+#   health.periodSeconds
+# Instead, please use the health.liveness and health.readiness specific properties
 health:
   path: '/health'
   initialDelaySeconds: 20
-  periodSeconds: 20
+  periodSeconds: 10
```
Comment (paraplexed) [inline charts/service/values.yaml:60]: Why are we reducing the default for this from 20 seconds to 10?

File: charts/service/templates/_probes.tpl (hunk 2 above)
Comment (ryan-hansen) [reply charts/service/templates/_probes.tpl:5]: My thinking here was the "default" would be the old way.  Only if they have liveness/readiness defined would it use those values.  We're gonna have to change the chart anyway as we deprecate the old method, so it basically equates to the same thing.  But I'm happy to swap it.  Doesn't matter to me.

File: charts/service/values.yaml (hunk 3 above)
Comment (ryan-hansen) [reply charts/service/values.yaml:60]: Negligence on my part.  Left over from some testing that I forgot to fix (which is where that chart.yaml file came from too)

**Pull request merged**